src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,519 @@
+    updateRegionConnection = QMetaObject::Connection();
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_sequenceModified() {
+    ADVSequenceObjectContext *sequenceContext = annDnaView->getActiveSequenceContext();
+    CHECK(sequenceContext != nullptr, )
+    qint64 seqLength = sequenceContext->getSequenceLength();
+
+    // End spinbox maximum value is the previous sequence length. If it has changed, need to update spinboxes.
+    if (seqLength != sbLeftAreaEnd->maximum()) {
+        int start = sbLeftAreaStart->value() - 1;
+        int len = sbLeftAreaEnd->value() - start;
+        setRegion(sbLeftAreaStart, {start, len});
+
+        start = sbRightAreaStart->value() - 1;
+        len = sbRightAreaEnd->value() - start;
+        setRegion(sbRightAreaStart, {start, len});
+    }
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_start() {
+    PCRPrimerDesignForDNAAssemblyTaskSettings settings;
+    settings.forwardUserPrimer = leForwardPrimer->text();
+    settings.reverseUserPrimer = leReversePrimer->text();
+
+    settings.gibbsFreeEnergy.minValue = sbMinRequireGibbs->value();
+    settings.gibbsFreeEnergy.maxValue = sbMaxRequireGibbs->value();
+    settings.meltingPoint.minValue = spMinRequireMeltingTeml->value();
+    settings.meltingPoint.maxValue = spMaxRequireMeltingTeml->value();
+    settings.overlapLength.minValue = spMinRequireOverlapLength->value();
+    settings.overlapLength.maxValue = spMaxRequireOverlapLength->value();
+
+    settings.gibbsFreeEnergyExclude = sbExcludeGibbs->value();
+    settings.meltingPointExclude = spExcludeMeltingTeml->value();
+    settings.complementLengthExclude = spExcludeComplementLength->value();
+
+    if (backbone5->isChecked()) {
+        settings.insertTo = PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone5;
+    } else {
+        settings.insertTo = PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone3;
+    }
+    settings.bachbone5Length = sbBackbone5Length->value();
+    settings.bachbone3Length = sbBackbone3Length->value();
+
+    settings.leftArea.startPos = (int)(sbLeftAreaStart->value()) - 1;
+    settings.leftArea.length = (int)sbLeftAreaEnd->value() - (int)sbLeftAreaStart->value();
+    settings.rightArea.startPos = (int)(sbRightAreaStart->value()) - 1;
+    settings.rightArea.length = (int)sbRightAreaEnd->value() - (int)sbRightAreaStart->value();
+
+    settings.backboneSequenceUrl = leBackboneFilePath->text();
+
+    settings.otherSequencesInPcrUrl = leOtherSequencesInPcrFilePath->text();
+
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    U2SequenceObject* sequenceObject = activeSequenceContext->getSequenceObject();
+    SAFE_POINT(NULL != sequenceObject, L10N::nullPointerError("Sequence Object"), );
+
+    U2OpStatus2Log os;
+    auto sequence = sequenceObject->getWholeSequenceData(os);
```

Comment:
Is there an example of such thing?

Reply:
Any genome sequencer results or whole genome sequences like http://hgdownload.cse.ucsc.edu/goldenpath/hg19/bigZips/

Reply:
I am able to find a large sequence) I mean, the example of the limitation code

Reply:
I see the "limitation" code here: `sequenceObject->getWholeSequenceData(os);`

If the sequence object is big enough and you are searching for primers in some limited 1k area the whole sequence will be copied. For 2GB+ data it simply won't work. For smaller sequences it will work inefficiently. We have a special class for that SequenceWalker that processes sequence of any size by chunks.

You can avoid the complex code of SequenceWalker by enabling the new plugin functionality only small sequences (up to several megabytes)